From 9db44754b9e4044690a6c32fd74b9d5fe26b07b2 Mon Sep 17 00:00:00 2001
From: Sergey Biryukov <sergeybiryukov.ru@gmail.com>
Date: Mon, 14 Oct 2019 10:52:04 +0000
Subject: [PATCH] Formatting: In `wp_validate_redirect()`, normalize the path
 when validating the location for relative URLs, to account for Windows paths.

Props peterwilsoncc, rconde, jmmathc, mat-lipe, Sixes, justinahinon, cmagrin, daxelrod, SergeyBiryukov.
Fixes #47980.
Built from https://develop.svn.wordpress.org/trunk@46472


git-svn-id: http://core.svn.wordpress.org/trunk@46270 1a063a9b-81f0-0310-95a4-ce76da25c4cd
---
 wp-includes/pluggable.php | 1 +
 wp-includes/version.php   | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/wp-includes/pluggable.php b/wp-includes/pluggable.php
index 468e06558668..206761aaef95 100644
--- a/wp-includes/pluggable.php
+++ b/wp-includes/pluggable.php
@@ -1418,6 +1418,7 @@ function wp_validate_redirect( $location, $default = '' ) {
 			$path = '';
 			if ( ! empty( $_SERVER['REQUEST_URI'] ) ) {
 				$path = dirname( parse_url( 'http://placeholder' . $_SERVER['REQUEST_URI'], PHP_URL_PATH ) . '?' );
+				$path = wp_normalize_path( $path );
 			}
 			$location = '/' . ltrim( $path . '/', '/' ) . $location;
 		}
diff --git a/wp-includes/version.php b/wp-includes/version.php
index f08acb103a41..39b23db8c1db 100644
--- a/wp-includes/version.php
+++ b/wp-includes/version.php
@@ -13,7 +13,7 @@
  *
  * @global string $wp_version
  */
-$wp_version = '5.3-beta3-46471';
+$wp_version = '5.3-beta3-46472';
 
 /**
  * Holds the WordPress DB revision, increments when changes are made to the WordPress DB schema.
